src/BehaviorsSDKManaged/Microsoft.Xaml.Interactions.Shared/Media/PlaySoundAction.cs

```diff
@@ -100,48 +112,67 @@ public object Execute(object sender, object parameter)
 			if (!Uri.TryCreate(this.Source, UriKind.Absolute, out sourceUri))
 			{
 				// Impose ms-appx:// scheme if user has specified a relative URI
-				string absoluteSource = string.Format(CultureInfo.InvariantCulture, PlaySoundAction.MsAppXSchemeFormatString, this.Source);
+				string absoluteSource = string.Format(CultureInfo.InvariantCulture, MsAppXSchemeFormatString, this.Source);
 				if (!Uri.TryCreate(absoluteSource, UriKind.Absolute, out sourceUri))
 				{
 					return false;
 				}
 			}
 
-			this._popup = new Popup();
-			MediaElement mediaElement = new MediaElement();
+			_popup = new Popup();
+            if (sender is UIElement element && element.XamlRoot != null)
+            {
+                _popup.XamlRoot = element.XamlRoot;
+            }
+
+            MediaPlayerElement mediaElement = new MediaPlayerElement();
 			_popup.Child = mediaElement;
 
 			// It is legal (although not advisable) to provide a video file. By setting visibility to collapsed, only the sound track should play.
 			mediaElement.Visibility = Visibility.Collapsed;
-			mediaElement.Source = sourceUri;
-			mediaElement.Volume = this.Volume;
 
-			mediaElement.MediaEnded += this.MediaElement_MediaEnded;
-			mediaElement.MediaFailed += this.MediaElement_MediaFailed;
+            mediaElement.Source = MediaSource.CreateFromUri(sourceUri);
+            mediaElement.AutoPlay = true;
+            mediaElement.MediaPlayer.Volume = this.Volume;
+            mediaElement.MediaPlayer.MediaEnded += this.MediaElement_MediaEnded;
+            mediaElement.MediaPlayer.MediaFailed += this.MediaPlayer_MediaFailed;
 
             this._popup.IsOpen = true;
 			return true;
 		}
 
-		private void MediaElement_MediaFailed(object sender, ExceptionRoutedEventArgs e)
+        private void MediaPlayer_MediaFailed(MediaPlayer sender, MediaPlayerFailedEventArgs args)
 		{
-			if (this._popup != null)
-			{
-				this._popup.IsOpen = false;
-				this._popup.Child = null;
-				this._popup = null;
-			}
+            // TODO: We should probably have some system/properties to report/bubble errors here
+            ClosePopup();
```

Comment:
We could expose a `MediaFailed` event from this behavior and raise that perhaps?

Reply:
Sure, I just left the TODO as it was a gap from the existing implementation. Do you want me to file an issue for us to address later or add a new event/call here now (I figured that was outside the scope of the intent of just updating/unifying the code here)?

Reply:
Yeah we can just do this in a follow up, not needed for 3.0.0 🙂